bam-index/src/main/java/org/victorchang/IndexVersion.java

```diff
@@ -0,0 +1,26 @@
+package org.victorchang;
+
+public abstract class IndexVersion {
+    private IndexVersion() {
+    }
+
+    public static final IndexVersion LATEST = new IndexVersion() {
+        @Override
+        public String fileName(String ext) {
+            return "qname-version" + version() + "." + ext ;
```

Comment:
So the file name is going to be e.g. `qname-version0.0` and `qname-version0.1`? I'm wondering whether we want to use an actual extension, e.g. `.bgz` (see list here: https://github.com/samtools/htsjdk/blob/master/src/main/java/htsjdk/samtools/util/FileExtensions.java#L79)

I think it would also be nice to have a bit more descriptive names. So with those two suggestions, the new names would be:

* `qname.0` -> `qname.v1.records.bgz`
* `qname.1` -> `qname.v1.index.bgz`

What do people think?

Reply:
I agree that more self-describing names is always good. They will be hidden in a directory next to the BAM file anyway, so we don't expect people to type them.

Reply:
yes we should!

Reply:
Small change

- `qname.0` -> `qname.v1.data.bgz`
- `qname.1` -> `qname.v1.index.bgz`

Reply:
Sounds good! That will also help my poor brain figure out which one comes first :)

Reply:
Minor: put version in the end? eg: `qname.data.v1.bgz` etc

Reply:
Readme also needs to be updated since it refers to the index file names.